File: backend/Proxies/RealDiscountService.cs
using System;
using System.Threading;

public class RealDiscountService : IDiscountService
{
    public double ApplyDiscount(double originalPrice, string discountCode)
    {
        Console.WriteLine("[RealDiscountService] Verificare si aplicare cupon în baza de date ...");
        
        if (discountCode == "VIP20")
        {
            Console.WriteLine("[RealDiscountService] Reducere de 20% aplicata cu succes.");
            return originalPrice * 0.80; 
        }

        Console.WriteLine("[RealDiscountService] Cod invalid.");
        return originalPrice;
    }
}

File: frontend/src/utils/proxy/RealDiscountService.js
import IDiscountService from './IDiscountService';

class RealDiscountService extends IDiscountService {
    applyDiscount(originalPrice, discountCode) {
        console.log("[RealDiscountService] Verificare si aplicare cupon...");
        
        if (discountCode === "VIP20") {
            console.log("[RealDiscountService] Reducere de 20% aplicata cu succes.");
            return { success: true, price: originalPrice * 0.80, message: "VIP20 aplicat cu succes." }; 
        }

        if (discountCode === "SALE10") {
            console.log("[RealDiscountService] Reducere de 10% aplicata cu succes.");
            return { success: true, price: originalPrice * 0.90, message: "SALE10 aplicat cu succes." };
        }

        console.log("[RealDiscountService] Cod invalid sau fara reducere.");
        return { success: false, price: originalPrice, message: "Cod invalid." };
    }
}

export default RealDiscountService;
